### ISSUE TITLE
Python type lookup broken when C domain is used

### ISSUE THREAD

**Issue by lazka** [sphinx-doc/sphinx#6488]
**Describe the bug**

The pycairo docs build has started to fail after some recent sphinx upgrade: https://travis-ci.org/pygobject/pycairo/jobs/543179784#L938

**To Reproduce**

conf.py:
```python
extensions = ['sphinx.ext.intersphinx']
intersphinx_mapping = {'python': ('https://docs.python.org/3', None)}
```

index.rst:
```rst
.. c:function:: cfoo()

.. function:: foo()

    :param float bar: :class:`float`
```

`python3 -m sphinx -b html -n . _build`

Expected output: No warnings

Actual output:

`index.rst:5: WARNING: py:type reference target not found: float`

Using:

* Python 3.7.3
* Sphinx 2.1.1

**Comment by lazka** [comment]
It works with 2.0.1 and is broken with 2.1.0

**Comment by lazka** [comment]
Bisected to a990f9ff724bc04e43

**Comment by tk0miya** [comment]
Sorry for inconvenience. Will be fixed by #6192

**Comment by lazka** [comment]
Thanks!

**Comment by lazka** [comment]
btw, did you reference the wrong PR? That's the one causing the issue :)

**Comment by lazka** [comment]
ok, #6482, I can confirm that that fixes my issue.

**Comment by tk0miya** [comment]
oh, sorry.

**Comment by tk0miya** [comment]
Fixed by #6478.
Thank you for reporting.
